packages/ion_identity_client/lib/src/core/identity_storage/identity_storage.dart

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'dart:async';
+
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:ion_identity_client/src/auth/dtos/dtos.dart';
+import 'package:ion_identity_client/src/core/identity_storage/data_storage.dart';
+import 'package:ion_identity_client/src/core/types/user_token.c.dart';
+import 'package:rxdart/rxdart.dart';
+
+class IdentityStorage {
+  IdentityStorage({
+    required FlutterSecureStorage secureStorage,
+  })  : _tokenStorage = DataStorage<Authentication>(
```

Comment:
I'd use DI. You can create a private constructor with `_tokenStorage` and `_privateKeysStorage` as params and a factory constructor that creates it with default values.